pcdsdevices/pim.py

```diff
@@ -420,6 +420,66 @@ class PPMPowerMeter(BaseInterface, Device):
                                    'EPICS updates.')
 
 
+class PPM_R_PowerMeter(BaseInterface, Device):
+    """
+    Analog measurement tool for beam energy as part of the PPM assembly.
+
+    When inserted into the beam, the `raw_voltage` signal value should
+    increase proportional to the beam energy. A responsivity value
+    calibrated for each power meter in units of volts per watt
+    is used to calculate the actual energy in the following way:
+
+    `calibrated_mj` = 1000 * (Signal - Background) / (Responsivity * Beam_Rate)
+
+    Analogous to PPMPowerMeter, with different ways to calculate background voltage and calibrated energy
+    """
+
+    tab_component_names = True
+
+    responsivity = Cpt(PytmcSignal, ':RES', io='io', kind='normal', doc='Responsivity in  V/W, unique for every power meter.')
+
+    background_voltage = Cpt(PytmcSignal, ':BACK:VOLT', io='io', kind='normal', doc='Background voltage value used to calculate pulse energy.')
+
+    auto_background_reset = Cpt(PytmcSignal, ':BACK:RESET', io='io', kind='normal', doc='Set to reset auto background voltage collection.')
+    set_metadata(auto_background_reset, dict(variety='command-proc', value=1))
+
+    background_mode = Cpt(PytmcSignal, ':BACK:MODE', io='io', kind='normal', doc='Can be manual or passive. In manual mode, you can collect for a specified number of seconds. In passive mode, a buffer of automatically collected background voltages will be used to calculate the background voltage.')
+    set_metadata(background_mode, dict(variety='command-enum'))
```

Comment:
btw, `command-enum` makes the typhos screen display separate buttons for each option, arranged vertically. Just making sure you know, some people prefer the combobox but it's a matter of taste and I don't have a strong opinion.

Reply:
I was assuming that you needed command-enum for the combobox.

Reply:
These aren't documented super well, the only way to know is to flip through the decorators at https://github.com/pcdshub/typhos/blob/master/typhos/widgets.py